cpp/ql/src/Telemetry/Diagnostics.qll

```diff
@@ -0,0 +1,32 @@
+import cpp
+
+/**
+ * A syntax error.
+ */
+class SyntaxError extends CompilerError {
+  SyntaxError() { this.getTag().matches("exp_%") }
+}
+
+/**
+ * A cannot open file error.
+ * Typically this is due to a missing include.
+ */
+class CannotOpenFile extends CompilerError {
+  CannotOpenFile() { this.hasTag("cannot_open_file") }
+
+  string getIncludedFile() {
+    result = this.getMessage().regexpCapture("cannot open source file '([^']+)'", 1)
+  }
+}
+
+/**
+ * An undefined identifier error.
+ * Currently unused.
+ */
+class UndefinedIdentifier extends CompilerError {
+  UndefinedIdentifier() { this.hasTag("undefined_identifier") }
```

Comment:
There are a number of other `..._undefined_identifier` tags, shouldn't those be covered too?

Reply:
I've had a search through the EDG source code, and the other `ec_undefined_` errors are all at warning or remark level (e.g. `ec_undefined_preproc_id`)

Reply:
Ok, I think I'll just delete `UndefinedIdentifier` as it's unused and not worth creating a test for.